src/service.ts

```diff
@@ -141,7 +165,7 @@ const GRPC_ERROR_CODE_TO_HTTP = {
     code: 401,
     message: 'Unauthorized',
   },
-};
+} as {[index: number]: {code: number, message: string}};
```

Comment:
Why is an object rather than an array? The cast would be unnecessary if this was rewritten as:
```ts
const GRPC_ERROR_CODE_TO_HTTP : Array<GrpcResponse> = [ ... ]
```

I wonder if there is a code-mod to convert such an object to array.

Reply:
This is exposed as a static on the Service class, so changing it would be a breaking change :/  I'm open to fixing it, but hesitant to do so in this PR.

Reply:
The changed I proposed isn't semver major though.

Reply:
In JavaScript an array `const a = ['hi']` is pretty much equivalent to `const b = {0: 'hi'}`. The latter is missing the `length` property and doesn't inherit from `Array.prototype`, but is otherwise similar.

If the `length` or the prototype are not part of the public contract, then this change wouldn't be semver major.

Reply:
Actually, `length` was missing, so it cannot be part of the semver contract. The next question is whether users expect this object to be inheriting from `Object.prototype`.

Reply:
I'm okay with this happening in a follow-on if you leave a TODO behind or open an issue to convert this to an array.